x-pack/plugins/security_solution/public/entity_analytics/components/entity_store/hooks/use_entity_engine_status.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { UseQueryOptions } from '@tanstack/react-query';
+import { useQuery } from '@tanstack/react-query';
+import type { ListEntityEnginesResponse } from '../../../../../common/api/entity_analytics';
+import { useEntityStoreRoutes } from '../../../api/entity_store';
+
+export const ENTITY_STORE_ENGINE_STATUS = 'ENTITY_STORE_ENGINE_STATUS';
+
+interface Options {
+  disabled?: boolean;
+  polling?: UseQueryOptions<ListEntityEnginesResponse>['refetchInterval'];
+}
+
+export const useEntityEngineStatus = (opts: Options = {}) => {
+  // QUESTION: Maybe we should have an `EnablementStatus` API route for this?
```

Comment:
Lets decide before merging

Reply:
I think we should but also think it's tech debt